### PR TITLE
Trello Task 3 - GET /api/topics

### PR DESCRIPTION
**Pull request by husainbismil**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+describe("NCNews-Server Unit Tests", () => {
+
+    describe("1) GET /api/topics/", () => {
+
+        test("Responds with an array of objects, where the objects have at least two properties: slug & description", () => {
+
+            return request(app).get("/api/topics").expect(200).then((response) => {
+                const topics = response.body;
+
+                expect(topics).toBeInstanceOf(Array);
```
Comment (k-huggs) [inline __tests__/app.test.js]: No need to check that this will be an instance of an array

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const express = require("express");
+const c = require("../controller/controller.js");
+const app = express();
+
+
+// Parse JSON by default
+app.use(express.json());
```
Comment (k-huggs) [inline app/app.js]: You're not using this so please delete until it's being utilised

File: app/app.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+// Endpoints
+app.get("/api/topics", c.getApiTopics);
+
+// Error Handling
+
+app.all('/*', (req, res) => {
```
Comment (k-huggs) [inline app/app.js]: You've put this in, however, there are no tests for a 404 in your test file. Please add the test prior to implementation

File: controller/controller.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// c = controller, m = model, v = view, db = database
+const m = require("../model/model");
+
+// TODO: find out how error catching is meant to be, repeating .catch here seems not very DRY
+
+// /api/topics Endpoint
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((queryResult) => {
+        const responseArray = queryResult.rows;
+        response.status(200).send(responseArray);
```
Comment (k-huggs) [inline controller/controller.js]: You don't want to be sending this back as an array, it should be on a object with the key being what specifically it is

File: controller/controller.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// c = controller, m = model, v = view, db = database
+const m = require("../model/model");
+
+// TODO: find out how error catching is meant to be, repeating .catch here seems not very DRY
+
+// /api/topics Endpoint
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((queryResult) => {
```
Comment (k-huggs) [inline controller/controller.js]: semantic variable naming please, query result can be anything!

File: model/model.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// c = controller, m = model, v = view, db = database
+const db = require("../db/connection.js");
+const c = require("../controller/controller.js");
```
Comment (k-huggs) [inline model/model.js]: You don't need to have the controller required into your model

File: package.json [hunk 7]
```diff
@@ -21,14 +21,17 @@
...
-    "pg-format": "^1.0.4",
-    "husky": "^7.0.0"
+    "pg-format": "^1.0.4"
   },
   "dependencies": {
-    "dotenv": "^16.0.0",
-    "pg": "^8.7.3"
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "lodash": "^4.17.21",
```
Comment (k-huggs) [inline package.json]: lodash has been installed but, what is it purpose currently?

**Review by k-huggs — COMMENTED** [review]
A few changes to make, have a look at the comments and resubmit :)

**Comment by husainbismil** [comment]
> A few changes to make, have a look at the comments and resubmit :)

changes made, thank you!

File: controller/controller.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+// c = controller, m = model, v = view, db = database
+const m = require("../model/model");
+
+// TODO: find out how error catching is meant to be, repeating .catch here seems not very DRY
+
+// /api/topics Endpoint
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((selectTopicsQueryResult) => {
+        const responseObject = {topics: selectTopicsQueryResult.rows};
```
Comment (k-huggs) [inline controller/controller.js]: The controller should just been handling the req and res. Any other manipulation should be performed in your model

**Review by k-huggs — APPROVED** [review]
One small change then merge 👍

**Pull request merged**